presto-main/src/test/java/com/facebook/presto/TestRequestHeaderModifier.java

```diff
@@ -0,0 +1,1211 @@
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.HttpUpgradeHandler;
+import javax.servlet.http.Part;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.security.Principal;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.EventListener;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestRequestHeaderModifier
+{
+    private MockWebServer mockWebServer;
+    private HttpServletResponse response;
+    private FilterChainStub filterChain;
+    private AuthenticationFilter filter;
+    private AuthenticatorStub authenticator;
+    private RequestModifierManagerStub requestModifierManager;
+    private RequestModifierStub requestModifier;
+
+    @BeforeMethod
+    public void setUp() throws IOException
+    {
+        mockWebServer = new MockWebServer();
+        mockWebServer.start();
+
+        response = new ConcreteHttpServletResponse();
+        filterChain = new FilterChainStub();
+
+        authenticator = new AuthenticatorStub();
+        requestModifierManager = new RequestModifierManagerStub();
+        requestModifier = new RequestModifierStub();
+
+        List<Authenticator> authenticators = Collections.singletonList(authenticator);
+        filter = new AuthenticationFilter(authenticators, new SecurityConfigStub(), requestModifierManager);
+    }
+
+    @AfterMethod
+    public void tearDown() throws IOException
+    {
+        mockWebServer.shutdown();
+    }
+
+    @Test
+    public void testDoFilter_SuccessfulAuthenticationWithHeaderModification() throws ServletException, IOException
+    {
+        mockWebServer.enqueue(new MockResponse().setBody("Mocked Body").setResponseCode(200));
+
+        ConcreteHttpServletRequest request = new ConcreteHttpServletRequest();
+        request.setPathInfo("/oauth2/token-value/");
+        request.setSecure(true);
+
+        PrincipalStub testPrincipal = new PrincipalStub();
+        authenticator.setPrincipal(testPrincipal);
+
+        requestModifierManager.setModifiers(Collections.singletonList(requestModifier));
+        requestModifier.setHeaderNames(Collections.singletonList("Extra-credential"));
+        requestModifier.setExtraHeaders(Collections.singletonMap("X-Custom-Header", "CustomValue"));
+
+        filter.doFilter(request, response, filterChain);
+
+        HttpServletRequest wrappedRequest = (HttpServletRequest) filterChain.getCapturedRequest();
+        assertEquals("CustomValue", wrappedRequest.getHeader("X-Custom-Header"));
+    }
+
+    abstract static class HttpServletRequestAdapter
```

Comment:
There's already a `MockHttpServletRequest`, can you use that?

Reply:
@tdcmeehan 
I have refactored the code to use MockHttpServletRequest. Could you check it?
Thanks.